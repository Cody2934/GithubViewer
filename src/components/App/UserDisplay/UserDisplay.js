import React from 'react';
import PropTypes from 'prop-types';

const UserDisplay = ({ username, profilePic, profileLink, repoLink, followerCount, followingCount }) => (
  <div>
    {username}
    {profilePic}
    {profileLink}
    {repoLink}
    {followerCount}
    {followingCount}
  </div>
);

UserDisplay.propTypes = {
  username: PropTypes.string.isRequired,
  profilePic: PropTypes.string.isRequired,
  profileLink: PropTypes.string.isRequired,
  repoLink: PropTypes.string.isRequired,
  followerCount: PropTypes.number.isRequired,
  followingCount: PropTypes.number.isRequired
};

export default UserDisplay;
