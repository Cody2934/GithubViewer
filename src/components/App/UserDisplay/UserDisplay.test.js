import React from 'react';
import { shallow } from 'enzyme';
import UserDisplay from './UserDisplay';

describe('UserDisplay', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<UserDisplay username="Cody"
      profilePic="this is a pic" 
      profileLink="this is a link" 
      repoLink="this is also a link" 
      followerCount={20}
      followingCount={30}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
