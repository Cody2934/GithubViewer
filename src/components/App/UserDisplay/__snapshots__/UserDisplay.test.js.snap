// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDisplay matches a snapshot 1`] = `
<div>
  Cody
  this is a pic
  this is a link
  this is also a link
  20
  30
</div>
`;
